mkdocs/docs/meili/v0.0.3/orchestration_thoughts.md

```diff
@@ -0,0 +1,53 @@
+# NPM Orchestration
+
+So for npm orchestration, there's a few design philosophies that are pivotal before we can even consider what kind of technology we're going to use
+
+## Design Philosophies
+
+### 1) Reproducible and hermetic
+Reproducible is a given since Cloesce's output is deterministic. What we really need to implement is making sure npm orchestration is hermetic. Builds shouldn't depend on the host machine's quirks and no matter what OS, PATH variables, global installs, when we install the package on a random device, ideally everything should behave the same.
+
+### 2) Cross-platform
+Goes hand in hand with 1, it should work across macOS/Linux/Windows without any weird shell quirks
+
+### 3) Speed
+We should aim for orchestration to be reasonably fast. Usually npm packages will cache things like configuration flags, model files if not changed, things like that so after the first run the npm command will run significantly faster, and since one of the big use cases for Cloesce is to advance vibe coding beyond just localhost, I assume users will be running cloesce over and over.
```

Comment:
absolutely, post MVP I want to explore shifting Cloesce to be an incremental compiler for maximum build speed in development